docs/design/generics/details.md

```diff
@@ -1335,13 +1334,60 @@ Carbon:
 
 ```carbon
 interface SetAlgebra {
-  extend Equatable;
-  extend ExpressibleByArrayLiteral;
+  extend require impls Equatable;
+  extend require impls ExpressibleByArrayLiteral;
 }
 ```
 
-> **TODO:** Document `extend [final] impl as I`, as adopted in
-> [p5337: Interface extension and `final impl` update](/proposals/p5337.md).
+With `extend require impls I`, an interface requires an `impl` to exist for `I`
+when implementing the containing interface. This requires writing at least two
+`impl`s to implement the containing interface. When the two interfaces are
+tightly coupled, it's possible to have one interface gain the members of and
+_provide_ an implementation for the other interface instead, with
+`extend impl as`.
+
+Just as when used in a class, the implied `Self` between `extend impl` and `as`
+must be omitted.
+
+When an interface `A` contains `extend impl as B`, implementing `A` will require
+writing an implementation of all members of `B` inside the `impl` of `A`. And
+anything that implements `A` will implicitly also implement `B`, using the
+definitions from the `impl` of `A`. Here is an example:
```

Comment:
The `A` and `B` in this paragraph are opposite how they are used in the example. Please change one so they match.